### PR TITLE
Changes to the test harness

### PR DESCRIPTION
**Pull request by mithro**
Making the test-runner run all the tests in parallel and use XHR requests.

### CODE REVIEW

**Comment by dstoc** [comment]
I'm surprised this runs faster given that the tests still run on the same thread.

Is it just that we're able to load the test resources in parallel?

**Comment by mithro** [comment]
I didn't check to see if the XHR change or the parallel tests makes it run faster.

**Comment by mithro** [comment]
Old

```
Ran 3032 tests in 25.978s
OK

real    0m26.737s
```

New

```
Ran 3032 tests in 12.327s
OK

real    0m13.588s
```

Most of the time now is left in starting up / stopping Chrome.

File: test/test-runner.html [hunk 1]
```diff
@@ -19,205 +21,256 @@
...
+  color: green;
+}
 
-var autoAdvance = true;
 
-var currentTest = 0;
-var currentTestStart;
+/* Pass styles */
+tbody.no-tests {
+  color: yellow;
```
Comment (dstoc) [inline test/test-runner.html:45]: Yellow is hard to read, maybe orangered?

**Comment by dstoc** [comment]
Can you add something so that when I run the test-runner manually I can easily know when it's finished?

**Comment by dstoc** [comment]
Need to add back the measured run time for each test.

**Comment by mithro** [comment]
Time measurements make no sense when the tests are run in parallel?

**Comment by dstoc** [comment]
It shows us how long a test takes to run relative to other tests.

**Comment by mithro** [comment]
But a test could be blocked on another test, they are no longer independent
if run in parallel?

**Pull request closed without merging**